### PR TITLE
Change `io::detail::orc` namespace into `io::orc::detail`

### PR DESCRIPTION
**Pull request by ttnghia**
This PR simply changes the (sub) namespace `io::detail::orc` namespace into `io::orc::detail`. From now, the implementation of ORC reader and writer will be part of the `detail` namespace under the `io::orc` namespace.

### CODE REVIEW

File: cpp/include/cudf/io/orc.hpp [hunk 1]
```diff
@@ -1304,7 +1304,7 @@ class orc_chunked_writer {
   void close();
 
   /// Unique pointer to impl writer class
-  std::unique_ptr<cudf::io::detail::orc::writer> writer;
+  std::unique_ptr<cudf::io::orc::detail::writer> writer;
```
Comment (vuule) [inline cpp/include/cudf/io/orc.hpp:1307]: How about this?
```suggestion
  std::unique_ptr<orc::detail::writer> writer;
```

File: cpp/src/io/orc/reader_impl.hpp [hunk 2]
```diff
@@ -34,7 +34,7 @@
 #include <utility>
 #include <vector>
 
-namespace cudf::io::detail::orc {
+namespace cudf::io::orc::detail {
 using namespace cudf::io::orc;
```
Comment (vuule) [inline cpp/src/io/orc/reader_impl.hpp:38]: We should not need this anymore, since now `cudf::io::orc::detail` is nested in `cudf::io::orc`.

**Review by hyperbolic2346 — APPROVED** [review]
❤️

**Review by mythrocks — APPROVED** [review]
Ah, this is much better.

File: cpp/src/io/orc/writer_impl.hpp
Comment (vuule) [inline cpp/src/io/orc/writer_impl.hpp:1]: L357 and L358 have fully qualified namespace - not needed any more.

File: cpp/src/io/functions.cpp [hunk 3]
```diff
@@ -270,7 +270,7 @@ void write_csv(csv_writer_options const& options,
     mr);
 }
 
-namespace detail_orc = cudf::io::detail::orc;
+namespace orc_detail = cudf::io::orc::detail;
```
Comment (vuule) [inline cpp/src/io/functions.cpp:273]: this alias only saves a single character - let's remove and use `orc::detail`

File: cpp/src/io/functions.cpp
Comment (vuule) [inline cpp/src/io/functions.cpp:1]: line 325 does not need `cudf::io` namespace

**Review by vuule — COMMENTED** [review]
Found a few more redundancies
looks great otherwise

**Pull request merged**